packages/nextjs/src/server/utils/requestResponseUtils.ts

```diff
@@ -0,0 +1,60 @@
+import { NextRequest } from 'next/server';
+
+import { AUTH_RESULT, RequestLike } from '../types';
+
+type RequestProps = {
+  req: NextRequest;
+  authResult: string;
+};
+
+export function setAuthResultOnRequest({ req, authResult }: RequestProps): void {
+  req.headers.set(AUTH_RESULT, authResult);
+  req.nextUrl.searchParams.set(AUTH_RESULT, authResult);
+}
+
+// Tries to extract auth result from the request using several strategies
+export function getAuthResultFromRequest(req: RequestLike): string | null | undefined {
+  return getQueryParam(req, AUTH_RESULT) || getHeader(req, AUTH_RESULT);
+}
+
+function getQueryParam(req: RequestLike, name: string): string | null | undefined {
+  if (req instanceof NextRequest) {
+    return req.nextUrl.searchParams.get(name);
+  }
+
+  // Check if the request contains a parsed query object
+  // NextApiRequest does, but the IncomingMessage in the GetServerSidePropsContext case does not
+  if ('query' in req) {
```

Comment:
🔧 This clause should check if `req.query[name]` is present. If it's not, shouldn't we try to fallback to `req.url` instead of returning `undefined`?

Reply:
The way `next` currently works the `query` is always up-to-date, when it exists.

The `req.url` case is mainly for the GSSP case where no `req.query` exists.

Reply:
That being said, I'll add the fallback there as well.